### PR TITLE
initial LLVM/Clang support

### PR DESCRIPTION
**Pull request by alfredfo**
This depends on compiler-rt support for cross compilation in another PR.

Need some input on this: https://github.com/gentoo/crossdev/compare/master...alfredfo:crossdev:crossdev-llvm?expand=1#diff-962cfe028826c645946cd4075fffa391761e468a3d08ac7f079d9bbf267dcd94R1698

Not super important: https://github.com/gentoo/crossdev/compare/master...alfredfo:crossdev:crossdev-llvm?expand=1#diff-a36adbe09874217b2001d0967fc567607eecf614ee7e2c278d2eff7551e5be6fR115

See: https://github.com/gentoo/gentoo/pull/32136

### CODE REVIEW

File: crossdev [hunk 1]
```diff
@@ -1610,13 +1668,44 @@ export EMERGE_DEFAULT_OPTS="--quiet-build=n"
 # screw random strictness in cross-compilers
 export FEATURES="${FEATURES} -stricter"
 
+if [[ "${LLVM}" == "yes" ]]; then
+    CLANG_CROSS_CFG_DIR=/etc/clang/cross
+    [ -d "${CLANG_CROSS_CFG_DIR}" ] || mkdir -p "${CLANG_CROSS_CFG_DIR}"
+
+    export CLANG_CROSS_CFG="${CLANG_CROSS_CFG_DIR}/${CTARGET}.cfg"
+    # Force --unwindlib=none for now
+    cat <<EOF > "${CLANG_CROSS_CFG}"
```
Comment (thesamesam) [inline crossdev:1677]: You can indent the contents if you include `-`:
```suggestion
    cat <<EOF- > "${CLANG_CROSS_CFG}"
```

File: crossdev [hunk 2]
```diff
@@ -1610,13 +1668,44 @@ export EMERGE_DEFAULT_OPTS="--quiet-build=n"
 # screw random strictness in cross-compilers
 export FEATURES="${FEATURES} -stricter"
 
+if [[ "${LLVM}" == "yes" ]]; then
+    CLANG_CROSS_CFG_DIR=/etc/clang/cross
+    [ -d "${CLANG_CROSS_CFG_DIR}" ] || mkdir -p "${CLANG_CROSS_CFG_DIR}"
```
Comment (thesamesam) [inline crossdev:1673]: We're in Bash, so let's consistently use Bash tests please:
```suggestion
    [[ -d "${CLANG_CROSS_CFG_DIR}" ]] || mkdir -p "${CLANG_CROSS_CFG_DIR}"
```

File: crossdev [hunk 3]
```diff
@@ -1667,8 +1764,8 @@ if ! ex_fast ; then
 	fi
 
 	# stage4: full compiler (C/C++/etc...)
-	if is_s4 ; then
-		EOPTS="${EOPTS_UP} --newuse"
+	if is_s4 && [[ ! "${LLVM}" == "yes" ]] ; then
```
Comment (thesamesam) [inline crossdev:1767]:
```suggestion
	if is_s4 && [[ "${LLVM}" != "yes" ]] ; then
```

File: crossdev [hunk 4]
```diff
@@ -123,6 +124,12 @@ STAGE_DISP=(
 
 parse_target() {
 	CTARGET=${1#cross-}
+    if [[ ${CTARGET} == "cross-*" ]] ; then
+        CTARGET=${1#cross-}
```
Comment (thesamesam) [inline crossdev:128]: Indentation looks inconsistent here (not tabbed)

**Review by thesamesam — COMMENTED** [review]
Just had a quick skim again, not an indepth review yet

Also, could any of this be split into a few commits? Or no?

**Comment by alfredfo** [comment]
> Just had a quick skim again, not an indepth review yet
Fixed them all now, thanks!

> Also, could any of this be split into a few commits? Or no?
Not sure if it makes any sense since pretty much all changes depend on each other.

**Comment by vimproved** [comment]
Any plans to implement flags for compiler-rt version/overlay/env?

**Comment by alfredfo** [comment]
> Any plans to implement flags for compiler-rt version/overlay/env?

Yes, adding that now

**Comment by alfredfo** [comment]
> > Any plans to implement flags for compiler-rt version/overlay/env?
> 
> Yes, adding that now

should work now, sry for being slow AF :)

**Pull request closed without merging**